pennylane/pauli/dla/structure_constants.py

```diff
@@ -36,6 +37,7 @@ def _all_commutators(ops):
 def structure_constants(
```

Comment:
The error when we input matrices without setting matrix=True could be more informative:
```
n = 4
gens = [qml.X(i) @ qml.X(i+1) + qml.Y(i) @ qml.Y(i+1) + qml.Z(i) @ qml.Z(i+1) for i in range(n-1)]
g = qml.lie_closure(gens, matrix=True)
g.shape

qml.structure_constants(g)
```
We get `AttributeError: 'numpy.ndarray' object has no attribute 'pauli_rep'`
This could instead suggest to the user that if they want to work with matrices/arrays, they should use `matrix=True`

Reply:
Yes good point!
The fix was relatively easy :)
before:
```
>>> qml.structure_constants([np.eye(2)])
AttributeError: 'numpy.ndarray' object has no attribute 'pauli_rep'
```

```
>>> qml.structure_constants([np.eye(2)])
ValueError: Cannot compute adjoint representation of non-pauli operators. Received [array([[1., 0.],
       [0., 1.]])]. If you want to use matrices, use structure_constants(.., matrix=True)
```